dask/dataframe/core.py

```diff
@@ -2152,6 +2181,99 @@ def quantile(df, q):
+    """DataFrame covariance and pearson correlation.
+
+    Computes pairwise covariance or correlation of columns, excluding NA/null
+    values.
+
+    Parameters
+    ----------
+    df : DataFrame
+    min_periods : int, optional
+        Minimum number of observations required per pair of columns
+        to have a valid result.
+    corr : bool, optional
+        If True, compute the Pearson correlation. If False [default], compute
+        the covariance.
+    scalar : bool, optional
+        If True, compute covariance between two variables as a scalar. Only
+        valid if `df` has 2 columns.  If False [default], compute the entire
+        covariance/correlation matrix.
+    """
+    if min_periods is None:
+        min_periods = 2
+    elif min_periods < 2:
+        raise ValueError("min_periods must be >= 2")
+    prefix = 'corr' if corr else 'cov'
+    df = df._get_numeric_data()
+    name = '{0}-agg-{1}'.format(prefix, tokenize(df, min_periods, scalar))
+    if scalar and len(df.columns) != 2:
+        raise ValueError("scalar only valid for 2 column dataframe")
+    k = '{0}-chunk-{1}'.format(prefix, df._name)
+    dsk = dict(((k, i), (cov_corr_chunk, f, corr))
+               for (i, f) in enumerate(df._keys()))
+    dsk[(name, 0)] = (cov_corr_agg, list(dsk.keys()), df._pd, min_periods,
+                      corr, scalar)
+    dsk = merge(df.dask, dsk)
+    if scalar:
+        return Scalar(dsk, name)
+    return DataFrame(dsk, name, df._pd, (df.columns[0], df.columns[-1]))
+
+
+def cov_corr_chunk(df, corr=False):
+    """Chunk part of a covariance or correlation computation"""
+    mat = df.values
+    mask = np.isfinite(mat)
+    keep = np.bitwise_and(mask[:, None, :], mask[:, :, None])
+
+    x = np.where(keep, mat[:, None, :], np.nan)
+    sums = np.nansum(x, 0)
+    counts = keep.astype('int').sum(0)
+    cov = df.cov().values
+    dtype = [('sum', sums.dtype), ('count', counts.dtype), ('cov', cov.dtype)]
+    if corr:
+        m = np.nansum((x - sums/np.where(counts, counts, np.nan))**2, 0)
+        dtype.append(('m', m.dtype))
+
+    out = np.empty(counts.shape, dtype=dtype)
+    out['sum'] = sums
+    out['count'] = counts
+    out['cov'] = cov * (counts - 1)
+    if corr:
+        out['m'] = m
+    return out
+
+
+def cov_corr_agg(data, meta, min_periods=2, corr=False, scalar=False):
+    """Aggregation part of a covariance or correlation computation"""
+    data = np.concatenate(data).reshape((len(data),) + data[0].shape)
+    sums = np.nan_to_num(data['sum'])
+    counts = data['count']
+
+    totals = np.cumsum(sums, 0)
+    ns = np.cumsum(counts, 0)
+
+    ts = ns[:-1] * sums[1:] - counts[1:] * totals[:-1]
+    dens = ns[1:] * ns[:-1] * counts[1:]
+    dens = np.where(dens, dens, np.nan)
+    C = (np.nansum(data['cov'], 0) +
+         np.nansum(ts * ts.transpose((0, 2, 1)) / dens, 0))
+    C[ns[-1] < min_periods] = np.nan
+    nobs = np.where(ns[-1], ns[-1], np.nan)
```

Comment:
If the intermediate results are sizable, then how much of a pain would it be to share the small boundary elements and do some of the aggregation on the chunk side?